src/map/map_state.cpp

```diff
@@ -286,6 +286,9 @@ void update_unit_arrows(sys::state& state, display_data& map_data) {
 	}
 
 	for(auto map_army : state.world.in_army) {
+		if(!state.world.army_is_valid(map_army)) {
+			return;
```

Comment:
If the army possibly being invalid is a problem, this won't actually fix anything, because it can still become invalid after this check. The game is running in parallel with this function